test/hotspot/jtreg/serviceability/jvmti/vthread/ThreadStateTest/ThreadStateTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8307365
+ * @summary Exercise JvmtiThreadState creation concurrently with terminating vthreads
+ * @requires vm.continuations
+ * @modules java.base/java.lang:+open
+ * @run main/othervm/native -agentlib:ThreadStateTest ThreadStateTest
+ */
+
+import java.util.concurrent.*;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+public class ThreadStateTest {
+    private static final String agentLib = "ThreadStateTest";
+    static final int VTHREAD_COUNT = 64;
+
+    private static native void SetSingleSteppingMode(boolean enable);
+    private static native void SetMonitorContendedMode(boolean enable);
+
+    final Runnable FOO = () -> {
+        Thread.yield();
+    };
+
+    private void runTest() throws Exception {
+        int tryCount = 150;
+
+        // Force creation of JvmtiThreadState on vthread start.
+        SetMonitorContendedMode(true);
+
+        while (tryCount-- > 0) {
+            ExecutorService scheduler = Executors.newFixedThreadPool(8);
```

Comment:
Is it possible to configure pool using any of these parameters?
-Djdk.defaultScheduler.parallelism= 
 -Djdk.defaultScheduler.maxPoolSize=

Reply:
The reason I'm not using the default scheduler is not to control the size of the pool but to be able to control its shutdown, since the carriers going away was part of the ThreadsListHandle crash.